tests/unit/scripting.tcl

```diff
@@ -612,6 +612,18 @@ start_server {tags {"scripting"}} {
         assert_equal [r ping] "PONG"
     }
 
+    test {Timedout read-only scripts can be killed by SCRIPT KILL even when use pcall} {
+        set rd [redis_deferring_client]
+        r config set lua-time-limit 10
+        $rd eval {local f = function() while 1 do redis.call('ping') end end while 1 do pcall(f) end} 0
+        after 200
+        catch {r ping} e
+        assert_match {BUSY*} $e
```

Comment:
let's change this to a wait_for_condition to avoid timing issues on slow machines (and also get faster execution on fast ones)